README.md

```diff
@@ -1773,3 +1801,100 @@ will return this response:
     }
 }
 ```
+
+## HTTP Server Proxy
+
+The proxy different stores can be used to both store and read their stored requests
+The configuration, as already defined in [HTTP Server](#http-server), is as follows:
+
+```
+"proxy": { // proxy configuration
+    "<store_name>": { // proxy store configuration
+        "mode": "passthru" // proxy store mode
+    }
+}
+```
+
+| Key            | Value Type     | Value description                                                        |
+|----------------|----------------|--------------------------------------------------------------------------|
+| proxy          | JSON Object    | An object with the store names as keys and their configuration as values |
+| <store_name>   | JSON Object    | An object with the store configuration                                   |
+| mode           | string         | The mode the store runs on (see below)                                   |
+
+Store modes:
+
+| Value        | Description                                                                            |
+|--------------|----------------------------------------------------------------------------------------|
+| passthru     |  The request to the proxy store will be stored as it is without any further processing |
+
+
+### Write to proxy store
+
+Perform a request against the http server path `/proxywrite/<store_name>`.
+Where `<store_name>` is a key (store name) inside the `proxy` object in the configuration.
+The expected response will have either `200` status code and the used offset as body or another status and an error body.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxywrite/test",
+    "method": "POST",
+    "query_params": {
+        "some": "param"
+    },
+    "header": {
+        "X-My-Header": 0
+    },
+    "body": {
+        "post": {
+            "my": ["body", "here"]
+        }
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+    "statuscode": 200,
+    "body": {
+        "offset": 0
+    }
+}
+```
+
+### Read from proxy store
+
+Whatever request performed against the server path `/proxyread/<store_name>?offset=<offset>`.
+Where:
+- `<store_name>` is a key inside the `proxy` object in the server configuration, aka the proxy store name
+- `<offset>` represents the entry to be retrieved in the proxy store requests collection. If not provided, 0 is assumed.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxyread/test",
+    "method": "GET",
+    "query_params": {
+        "offset": 0
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+    "header": { // Merged headers. original request headers prefixed with 'X-Request`
+        "X-Apitest-Proxy-Request-Method": ["POST"], // The method of the request to the proxy store
+        "X-Apitest-Proxy-Request-Path": ["/proxywrite/test?is=here"], // The url path requested (including query string)
+        "X-Apitest-Proxy-Request-Query": ["is=here&my=data&some=value"], // The request query string only
```

Comment:
Lets stick to Go. Path is just the path, Query should be RawQuery and be the query